### DISCUSSION TITLE
TNoodle Scrambles

### DISCUSSION THREAD

**Discussion by waterflamev8** [thewca/tnoodle-lib#32] [Q&A]
Hi, I want to use tnoodle-lib to generate some scrambles. The 6x6 and 7x7 scrambles are not available so may I know how I should go about implementing them.

Thanks

**Comment by waterflamev8** [comment]
Does anyone look at discussions? 🤔

**Reply by gregorbg** [reply]
Nope, this is a new feature for us so we are not used to checking here. Thank you for pinging us and sorry for the late reply :grin: Typing the answer now...

**Comment by gregorbg** [comment]
6x6 and 7x7 are implemented through the generic `CubePuzzle` class. That class takes an `int` in its constructor, so:
- for 6x6 you do `new CubePuzzle(6)`
- for 7x7 you do `new CubePuzzle(7)`

As an alternative, you may use the [PuzzleRegistry](https://github.com/thewca/tnoodle-lib/blob/master/scrambles/src/main/java/org/worldcubeassociation/tnoodle/scrambles/PuzzleRegistry.java) class to access scrambler implementations for all WCA puzzles

**Reply by waterflamev8** [reply]
Ok thanks so much!
